unit_tests/save_data_tests/test_save_players.py

```diff
@@ -0,0 +1,88 @@
+import pytest
+from unittest.mock import patch
+from database.postgres import save_api_data
+from database.session import SessionLocal
+
+
+class TestPlayersSavingData:
+    @pytest.fixture(autouse=True)
+    def setup(self):
+        self.sport_name = "mma"
+
+
+    @patch('database.postgres.dal.PlayerDal.save_players')
+    def test_parameters_with_team(self, mock_save_players):
+        json_case_valid_parameters = {
+            "get": "players",
+            'parameters': {
+                'team': 1488
+            },
+            "response": [{}]
+        }
+
+        save_api_data(json_case_valid_parameters, self.sport_name)
+        mock_save_players.assert_called_once()
+
+
+    @patch('database.postgres.dal.PlayerDal.save_players')
+    def test_full_info(self, mock_save_players):
+        json_case_full_info = {
+            "get": "players",
+            'parameters': {},
+            "response": [
+                {
+                    "id": 6,
+                    "name": "Neymar",
+                    "photo": "https://media.api-sports.io/football/players/276.png",
+                    "team": [
+                        {
+                            "id": 4,
+                            "name": "SBG Ireland"
+                        }
+                    ]
+                },
+                {
+                    "id": 3,
+                    "name": "John Doe",
+                    "photo": "https://example.com/john_doe.png",
+                }
+            ]
+        }
+
+        save_api_data(json_case_full_info, self.sport_name)
+        mock_save_players.assert_called_once()
```

Comment:
Is there a way for us to also verify correct data being stored in fake DB?
"CalledOnce" is a very easy pass.

You can either try to test the data stired in the fake DB (would be nice, even though it's closer to integration testing) or check that `database.postgres.dal.PlayerDal.save_players` is invoked with a PSECIFIC set of parameters.

For example you expect team_index_id to equal 5 or smth and you expect the icon to be something. That kind of stuff. This way you still do UNIT testing, but in a proper way.

Basically, you make sure your `save_api_data` method processed info correctly and passed it to `database.postgres.dal.PlayerDal.save_players` in a proper way as well